dark/reads.py

```diff
@@ -133,6 +135,26 @@ def walkHSP(self, hsp):
             readOffset += 1
             subjectOffset += 1
 
+    def checkAlphabeth(self, count=10):
+        """
+        A function which checks whether the sequence in a L{dark.Read} object
+        corresponds to its readClass. For AA reads, more testing is done in
+        dark.Read.AARead.checkAlphabeth.
+
+        @param count: A C{int} of how many amino acids should be considered.
+        @return: C{True} if the sequence corresponds from its readClass or
+            C{False} if the sequence doesn't correspond.
+        @raise ValueError: If the sequence and the readClass don't match, raise
+            a ValueError.
+        """
+        readLetters = set(self.sequence.upper()[:count])
```

Comment:
That sounds ok. It's again the most conservative / slowest option. It's
like with FastaReads we check all reads, and we're checking all bases of
all reads.  I don't really like it, but at least it's consistent, and if
someone cares they can arrange to do less work.  One small efficiency thing
is that using `seq[:None]` results in a copy being made, so an alternative
would be to use `if count is None` and in that case just use `seq` (i.e.,
no copying). I guess I should just stop worrying about speed..... old
dinosaur habit.


On Sun, Apr 12, 2015 at 10:21 PM, Barbara Mühlemann <
notifications@github.com> wrote:

> In dark/reads.py
> <https://github.com/acorg/dark-matter/pull/318#discussion_r28207600>:
>
> > @@ -133,6 +135,26 @@ def walkHSP(self, hsp):
> >              readOffset += 1
> >              subjectOffset += 1
> >
> > +    def checkAlphabeth(self, count=10):
> > +        """
> > +        A function which checks whether the sequence in a L{dark.Read} object
> > +        corresponds to its readClass. For AA reads, more testing is done in
> > +        dark.Read.AARead.checkAlphabeth.
> > +
> > +        @param count: A C{int} of how many amino acids should be considered.
> > +        @return: C{True} if the sequence corresponds from its readClass or
> > +            C{False} if the sequence doesn't correspond.
> > +        @raise ValueError: If the sequence and the readClass don't match, raise
> > +            a ValueError.
> > +        """
> > +        readLetters = set(self.sequence.upper()[:count])
>
> So if you say
> seq[:None] you get the whole sequence. So I can just change the default
> of count to None.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/acorg/dark-matter/pull/318/files#r28207600>.
>